docs/source/user/conversion.rst

```diff
@@ -0,0 +1,97 @@
+.. _conversion:
+
+===============
+Data Conversion
+===============
+
+By default PyMySQL will convert MySQL data types into native Python types. The
+full list of conversions is:
+
+=============== =============================================
+MySQL type      Python type
+=============== =============================================
+``BIT``         [No conversion]
+``TINY``        ``Int``
+``SHORT``       ``Int``
+``LONG``        ``Int``
+``FLOAT``       ``Float``
+``DOUBLE``      ``Float``
+``LONGLONG``    ``Int``
+``INT24``       ``Int``
+``YEAR``        ``Int``
+``TIMESTAMP``   ``datetime.datetime`` or ``datetime.date`` *
+``DATETIME``    ``datetime.datetime`` *
+``TIME``        ``datetime.timedelta`` *
+``DATE``        ``datetime.date`` *
+``SET``         ``Set``
+``BLOB``        ``String``
+``TINY_BLOB``   ``String``
+``MEDIUM_BLOB`` ``String``
+``LONG_BLOB``   ``String``
+``STRING``      ``String``
+``VAR_STRING``  ``String``
+``VARCHAR``     ``String``
+``DECIMAL``     ``Decimal``
+``NEWDECIMAL``  ``Decimal``
+=============== =============================================
+
+Items marked * are returned as ``None`` if they are of an invalid format or are
+invalid dates. For example, these valid ``DATETIME`` values::
```

Comment:
I don't will wasting your work.  I just want strong message "don't use this, unless no other way!" in this document.

This feature is kept with many pitfalls and special dirty hacks.  This feature makes difficult to finding and fixing bugs without regression or backward incompatibility.

For example, see PyMySQL/mysqlclient-python#145 and PyMySQL/mysqlclient-python#155 are recent regression caused by basic design of converter.

Another example is this dirty code in https://github.com/PyMySQL/mysqlclient-python/blob/master/MySQLdb/connections.py
:

```
        db = proxy(self)
        def _get_string_literal():
            # Note: string_literal() is called for bytes object on Python 3 (via bytes_literal)
            def string_literal(obj, dummy=None):
                return db.string_literal(obj)
            return string_literal
```

converter (both of encoder and decoder) should know encoding of connection.
But converters doesn't have encoding argument.  So string encoder is overwritten by connection constructor.  And weakref is required to avoid circular reference (it cause leak at Python 2.  I now hate Python 2 too...).
Decoder has another dirty hack too.

If no one use converters, I can add new argument and pass necessary information via argument.  Then I'll be able to remove lots of unmaintainable dirty hacks!!!

I want to do it someday.  But it breaks backward compatibility.
That's why I don't want to people using it.